### PR TITLE
Shenandoah unified logging

### PR DESCRIPTION
**Pull request by cfumiwu**
Writing the Shenandoah log file to the log stream without creating a new stream for it.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Reviewers
 * [William Kemper](https://openjdk.org/census#wkemper) (@earthling-amzn - Committer) ⚠️ Review applies to [22dfc26a](https://git.openjdk.org/shenandoah/pull/162/files/22dfc26a3131192aaccc946effbfc3b6803c30be)
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - Committer) ⚠️ Review applies to [22dfc26a](https://git.openjdk.org/shenandoah/pull/162/files/22dfc26a3131192aaccc946effbfc3b6803c30be)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/shenandoah pull/162/head:pull/162` \
`$ git checkout pull/162`

Update a local copy of the PR: \
`$ git checkout pull/162` \
`$ git pull https://git.openjdk.org/shenandoah pull/162/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 162`

View PR using the GUI difftool: \
`$ git pr show -t 162`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/shenandoah/pull/162.diff">https://git.openjdk.org/shenandoah/pull/162.diff</a>

</details>

### CODE REVIEW

**Review by earthling-amzn — APPROVED** [review]
This looks great. Thank you.

File: src/hotspot/share/gc/shenandoah/shenandoahHeapRegionCounters.cpp [hunk 1]
```diff
@@ -75,17 +74,33 @@ ShenandoahHeapRegionCounters::ShenandoahHeapRegionCounters() :
...
+        ls.print_cr("%li %li %lu %lu %lu",
+                    ts->get_value(), status->get_value(), num_regions, region_size, protocol_version);
+        if (num_regions > 0) {
+            ls.print("%li", regions[0]->get_value());
+        }
+        for (uint i = 1; i < num_regions; ++i) {
+            ls.print(" %li", regions[i]->get_value());
+        }
+        ls.cr();
+    }
```
Comment (shipilev) [inline src/hotspot/share/gc/shenandoah/shenandoahHeapRegionCounters.cpp:103]: I think the indenting style is 2 spaces.

**Review by shipilev — APPROVED** [review]
Yes, see how UL is more convenient than ad-hoc logging system :)

File: src/hotspot/share/gc/shenandoah/shenandoahHeapRegionCounters.cpp (hunk 1 above)
Comment (earthling-amzn) [reply src/hotspot/share/gc/shenandoah/shenandoahHeapRegionCounters.cpp]: Thank you @shipilev . We have a PR open on the visualizer itself with changes that compliment these: https://github.com/openjdk/shenandoah-visualizer/pull/1. We would really appreciate it if you could find time to also review that PR.

**Pull request closed without merging**